### PR TITLE
Murori: Adds more datasets, including SUMMscreen, QMsum, Arxiv, XSum, PubmedQA

### PR DESCRIPTION
**Pull request by MuroriM**
Added SUMMscreen dataset and QMsum datasets.
These changes affect dataset/non_huggingface_datasets.py and dataset/__init__.py.

Made a few minor fixes in:
huggingface_datasets.py - to correct the types of certain SummInstance variables
tests/dataset_test.py - corrects to the correct implementation of checking whether SummInstance is a list or string

### CODE REVIEW

File: dataset/huggingface_datasets.py [hunk 1]
```diff
@@ -140,7 +139,7 @@ def __init__(self):
     @staticmethod
     def process_samsum_data(data: Dataset) -> List[SummInstance]:
         for instance in tqdm(data):
-            dialogue: str = instance['dialogue'].split('\r\n')  # split each dialogue into a list of strings such as
+            dialogue: List = instance['dialogue'].split('\r\n')  # split each dialogue into a list of strings such as
```
Comment (niansong1996) [inline dataset/huggingface_datasets.py:142]: Here you should say `List[str]` to be exact

File: dataset/non_huggingface_datasets.py [hunk 2]
```diff
@@ -3,7 +3,9 @@
 import os
 import json
 import random
-from urllib.request import urlretrieve
+import gdown
```
Comment (niansong1996) [inline dataset/non_huggingface_datasets.py:6]: Is gdown a new dependency? If so, you need to change the requirements.txt as well

**Review by niansong1996 — COMMENTED** [review]
Generally looks good to me, we can merge this if all tests are passing

**Comment by MuroriM** [comment]
Adds Arxiv and Xsum datasets. Still testing the Arxiv dataset due to its enormous size, so in case there's an error in the code, I'll push some more commits to fix this.

The goal is to have all commits in by tomorrow morning

**Review by niansong1996 — COMMENTED** [review]
Looks good to me, this can be merged if all tests are passing

**Comment by MuroriM** [comment]
Adds the Pubmed QA dataset to cover the medical domain

Modifies:
dataset/huggingface_datasets.py
dataset/__init__.py

**Comment by niansong1996** [comment]
Nice work! Are all five datasets done and equipped with tests?

**Pull request merged**